compositor_render/src/transformations/builtin.rs

```diff
@@ -28,20 +28,30 @@ impl Builtin {
                 background_color_rgba,
                 ..
             } => Some(rgba_to_wgpu_color(background_color_rgba)),
+            BuiltinSpec::MirrorImage { .. } => None,
         }
     }
 
-    pub fn output_resolution(&self, _input_resolutions: &[Option<Resolution>]) -> Resolution {
+    pub fn output_resolution(&self, input_resolutions: &[Option<Resolution>]) -> Resolution {
         match self.0 {
             BuiltinSpec::TransformToResolution { resolution, .. } => resolution,
             BuiltinSpec::FixedPositionLayout { resolution, .. } => resolution,
+            BuiltinSpec::MirrorImage { .. } => input_resolutions
+                .first()
+                .copied()
+                .flatten()
+                .unwrap_or(Resolution {
+                    width: 1,
+                    height: 1,
+                }),
```

Comment:
We should probably create sth like fallbacks, where each of those transformations would only specify foxed resolution or resolution of input or other similar options.